### PR TITLE
COCOS-144 - Add Agent SDK Tests

### LINKED ISSUE #144: Feature: Add SDK Tests
### Is your feature request related to a problem? Please describe.

Add tests on sdk package

### Describe the feature you are requesting, as well as the possible use case(s) for it.

Add tests on sdk package

### Indicate the importance of this feature to you.

Must-have

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by WashingtonKK**
# What type of PR is this?

This is an enhancement since it adds tests to sdk.

## What does this do?

This adds sdk tests to cocos.

## Which issue(s) does this PR fix/relate to?

- Resolves #144

## Have you included tests for your changes?

Yes, I have included tests for my changes.

## Did you document any new/modified feature?

Yes, I have updated the documentation for the new feature.

### CODE REVIEW

**Review by SammyOina — CHANGES_REQUESTED** [review]
coverage on agent.go is about 51.9% which is quite low. Add more test cases and check on coverage using 

```shell
go test ./... -coverprofile cover.out
go tool cover -html=cover.out  
```

File: pkg/sdk/agent_test.go [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+			err:     errors.New("missing algorithm"),
+			wantErr: true,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			svcCall := svc.On("Algo", mock.Anything, mock.Anything).Return(tc.err)
+			if err = sdk.Algo(context.Background(), tc.algo, tc.userKey); err == nil && tc.wantErr {
+				t.Errorf("expected error, got none")
```
Comment (SammyOina) [inline pkg/sdk/agent_test.go:109]: Use error contains instead

File: pkg/sdk/agent_test.go [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+			wantErr: true,
+		},
+		{
+			name: "missing algorithm",
+			algo: agent.Algorithm{
+				UserKey: algorithmProviderPubKey,
+				Hash:    algoHash,
+			},
+			userKey: algorithmProviderKey,
+			err:     errors.New("missing algorithm"),
```
Comment (SammyOina) [inline pkg/sdk/agent_test.go:100]: use the errors exported from agent and wrap errors when required

**Review by SammyOina — CHANGES_REQUESTED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Comment by drasko** [comment]
@WashingtonKK please change according to @SammyOina 's comment and let's merge this one

**Review by SammyOina — APPROVED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**